### ISSUE TITLE
Could not use %PROGRAMFILES(x86)% shortcut in plugin settings

### ISSUE THREAD

**Issue by qiansen1386** [atom/atom#8790]
Greeting, Developers!

I used to add %PROGRAMFILES(x86)%\Graphviz2.38\bin\dot.exe to many plantuml plugins.
They all prompted me errors saying that the executable file does not exists.

It totally make sense to follow Windows convention in Windows Environment I believe.

Thanks.

**Comment by damieng** [comment]
Hi qiansen1386. You probably need to put double-quotes around that so that as %PROGRAMFILES(x86)% will contain spaces when it expands.

Do you still have this issue?

**Comment by qiansen1386** [comment]
@damieng Thanks u for ur reply. Let me try it first, will update u soon.

**Comment by qiansen1386** [comment]
Unfortunately, the double-quotes do not work man.
I am using Windows 7 64bit, is there a format or tool for us to export the necessary system info to help your troubleshooting?

**Comment by paulcbetts** [comment]
I don't think that Atom settings will expand environment variables, you probably have to use the real path

**Comment by qiansen1386** [comment]
Right, but the initial purpose of this post is exactly to ask for the
feature to support environment variables.
The problem:
Just imagine if JRE keep updating their JAVA_HOME path, and we could not
use the placeholder variable, then we will be forced to update them
whenever we upgraded JAVA. I think that is not so good for Windows users.

On Mon, Feb 29, 2016 at 10:11 AM, Paul Betts <notifications@github.com>
wrote:

> I don't think that Atom settings will expand environment variables, you
> probably have to use the real path
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/atom/issues/8790#issuecomment-189996517>.
>



-- 

Regards,
Paris Qian Sen （钱森）

**Comment by damieng** [comment]
In what places do you want to use environment variables?  (Trying to do it absolutely everywhere would be problematic)

**Comment by qiansen1386** [comment]
When the plugins require some local installations. Like PlantUml => dot.exe or atom-beautify => local formatters.

**Comment by mickhrmweb** [comment]
I'd also absolutely love to be able to use env vars in configuration fields.  The [racer](https://atom.io/packages/racer) plugin requires absolute paths to files and directories and makes it painful to maintain a reusable configuration because of that.

**Comment by qiansen1386** [comment]
Please leave it open till it get solved.
